server/v2/core/store/types.go

```diff
@@ -6,27 +6,55 @@ import (
 
 type Hash = []byte
 
-var _ store.KVStore = (WritableState)(nil)
+var _ store.KVStore = (Writer)(nil)
 
 // Store defines the underlying storage engine of an app.
 type Store interface {
 	// StateLatest returns a readonly view over the latest
 	// committed state of the store. Alongside the version
 	// associated with it.
-	StateLatest() (uint64, ReadonlyState, error)
+	StateLatest() (uint64, GetReader, error)
 
 	// StateAt returns a readonly view over the provided
 	// state. Must error when the version does not exist.
-	StateAt(version uint64) (ReadonlyState, error)
+	StateAt(version uint64) (GetReader, error)
 
 	// StateCommit commits the provided changeset and returns
 	// the new state root of the state.
-	StateCommit(changes []ChangeSet) (Hash, error)
+	StateCommit(changes []AccountStateChanges) (Hash, error)
 }
 
-// ChangeSet represents a change in a key and value of state.
+// GetReader represents a readonly view over all the accounts state.
+type GetReader interface {
+	// GetAccountReadonlyState must return the state for the provided address.
+	// Storage implements might treat this as a prefix store over an address.
+	// Prefix safety is on the implementer.
+	GetAccountReader(address []byte) (Reader, error)
+}
+
+// WritableAccountsState represents a writable account state.
```

Comment:
```suggestion
// GetWriter represents a writable account state.
```

also, why is account mentioned everywhere? What does account mean in the server v2 `store` package?

Reply:
Account = "a system actor" in this case it holds state.

IDK of a better name

Reply:
happy to go with a different approach, we can do namespace too if we wanted. Its all similar to store key, can do that too

Reply:
I was confusing account with the accounts inside the SDK 🤦
So it's basically bank, auth, accounts, etc or is it at a different level than that? Like state machines (for when we have multiple state machines)?

Reply:
I consider this a blocker -- account is overloaded and will 100% cause confusion (as it already does). Perhaps we consider actor or something similar.

Reply:
Yeah my points still remains, `account` is poor nomenclature here and will cause confusion

Reply:
we can remove account and call it a namespace, it doesnt imply account nor specific store like store key.